course7/homeworks/fred/complex-rpn/main.cc

```diff
@@ -0,0 +1,65 @@
+#include "complex.h"
+#include "calculator.h"
+#include "utils.h"
+
+#include <iostream>
+#include <stdexcept>
+
+void insert(RpnCalculator& calculator, Complex complex) {
+  calculator.insert(complex);
+}
+
+void insert(RpnCalculator& calculator, Operator op) {
+  auto result = calculator.insert(op);
+  std::cout << "Result so far: " << result << std::endl;
+}
+
+int main(int argc, char** args) {
+  std::cout << "Welcome!" << std::endl;
+  std::cout << "Type 'q' to quit" << std::endl << std::endl;
+
+  RpnCalculator calculator;
+
+  std::string input = "";
+  while (true) {
+    std::getline(std::cin, input);
+    if (input == "q")
+      break;
+
+    bool parseFailed = false;
+
+    try {
+      // try to parse a number first
+      auto parseComplexResult = tryParseComplexFromString(input);
```

Comment:
The code would be much clearer if you'd have a function like ```isOperaror```.
The 'tryParse' functions do 2 things. It is always a good idea for function/class to do exactly 1 thing.